.github/workflows/build.yml

```diff
@@ -10,25 +10,24 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
       - name: Run golangci-lint
-        uses: golangci/golangci-lint-action@v2
+        uses: golangci/golangci-lint-action@v3
         with:
-          # Hard-coding version due to this bug: https://github.com/golangci/golangci-lint-action/issues/535
-          version: v1.47
+          version: v1.55.2
   test:
     name: go test
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
       - name: Set up Go
-        uses: actions/setup-go@v2
+        uses: actions/setup-go@v5
         with:
-          go-version: 1.18
+          go-version: 1.21.6
```

Comment:
Are you sure that you want to specify the patch version (`.6`)?  (If you omit it, doesn't that let it "float" to the latest?...which seems good and appropriate, to me.)

Ditto line 70.

Reply:
Right, because patch versions never break anything. Given some of our experiences (though we might wish for the Go ecosystem to be more robust than PyPi), I'm not sure that erring on the side of irrational paranoia is necessarily a bad thing!

Reply:
When I made these changes I thought the [golang 1.21 release docs](https://tip.golang.org/doc/go1.21#introduction) were saying that I needed to specify the patch version, but it looks like I do not. Also, @dbutenhof  has a point.

Reply:
> I'm not sure that erring on the side of irrational paranoia is necessarily a bad thing!

Yep.  I'm jes' tryin' to confirm that that's the poison that we want to pick:  we can either let it float a little bit and thereby avoid the pitfalls of missing patch versions and the pains of having to upgrade more often or we can nail it down and avoid the pain of being broken by an irresponsible or erroneous upgrade on their end, should that ever occur.  Like everything else we do, it's a trade-off, and it's not necessarily clear which way we should go.  (So, I thought I would make sure that Matt had given the choice its due consideration.)

Reply:
yeah we're going to leave it pinned at 1.21.6